lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,309 @@
+                                 course_id: this.currentSessionSelection,
+                                 course_uuid: this.entitlementModel.get('entitlementUUID')
+                             }
+                         }),
+                         success: _.bind(this.enrollSuccess, this),
+                         error: _.bind(this.enrollError, this)
+                     });
+                 },
+
+                 enrollSuccess: function(data) {
+                     /*
+                     Update external elements on the course card to represent the now available course session.
+
+                     On enroll:
+                     1) Show the change session toggle button.
+                     2) Add the new session's dates to the date field on the main course card.
+                     3) Hide the 'View Course' button to the course card.
+
+                     On unenroll:
+                     1) Hide the change session button and the date field.
+                     2) Hide the 'View Course' button.
+                     3) Remove the messages associated with the enrolled state.
+                     4) Remove the link from the course card image and title.
+                     */
+                     var enrolled = data.is_active;
+
+                     // Update the model with the new session Id;
+                     this.entitlementModel.set({currentSessionId: this.currentSessionSelection});
+                     if (enrolled) {
+                         this.$triggerOpenBtn.removeClass('hidden');
+                         this.$dateDisplayField.html(this.getAvailableSessionWithId(data.course_details.course_id)
+                             .session_dates).prepend('<span class="fa fa-check" aria-hidden="true"></span>');
+                         this.$enterCourseBtn.attr('href', this.formatCourseHomeUrl(data.course_details.course_id))
+                             .removeClass('hidden');
+                         this.toggleSessionSelectionPanel();
+                     } else {
+                         // Reset the card contents to the unenrolled state
+                         this.$triggerOpenBtn.addClass('hidden');
+                         this.$enterCourseBtn.addClass('hidden');
+                         this.$dateDisplayField.html('');
+                         this.$courseCardMessages.remove();
+
+                         // Remove links to previously enrolled sessions
+                         this.$courseImageLink.replaceWith('<div class="' + this.$courseImageLink.attr('class') +
+                             '" tabindex="-1">' + this.$courseImageLink.html() + '</div>');
+                         this.$courseTitleLink.replaceWith('<span>' + this.$courseTitleLink.text() + '</span>');
+                     }
+                 },
+
+                 enrollError: function() {
+                     this.$dateDisplayField.find('.fa.fa-spin').removeClass('fa-spin fa-spinner').addClass('fa-close');
```

Comment:
Style nitpick; it is common to take a new line for chained methods, which makes it easier to read. E.g.
```
this.$dateDisplayField
    .find('.fa.fa-spin')
    .removeClass('fa-spin fa-spinner')
    .addClass('fa-close');
```

Reply:
Changed